src/others/components/FilterDropdown/ListWithSearch.tsx

```diff
@@ -0,0 +1,113 @@
+import { FixedSizeList, areEqual } from "react-window";
+import { Button, ListItem, ListItemButton, ListItemText, ClickAwayListener, Divider, useAutocomplete, Typography } from "@mui/material";
+import { FunctionComponent, memo, forwardRef } from "react";
+import SearchIcon from "@mui/icons-material/Search";
+
+import { FilterItem } from "../../contexts/filter";
+
+import styles from "./ListWithSearch.module.css";
+
+type ListWithSearchProps = {
+  selectedFilterItemCount: number;
```

Comment:
**suggestion**

I don't think that's necessary, we should "derive state" where possible and minimise components' API. 
If `selectedFilterItemCount` is `selectedFilterItems.length`, we can easily create it within the component's body.